tests/syslog/test_syslog_source_ip.py

```diff
@@ -0,0 +1,555 @@
+
+    @pytest.fixture(scope="class", autouse=True)
+    def setup_ssip_test_env(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_frontend_asic_index,
+                            mgmt_interface, routed_interfaces, backup_and_restore_config_db_on_duts):
+        """
+        Setup env for ssip(syslog soruce ip) test
+        """
+        self.duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+        self.asichost = self.duthost.asic_instance(enum_frontend_asic_index)
+
+        logger.info("Configure IP on the selected interface in default vrf")
+        self.configure_default_vrf_test_data(routed_interfaces)
+
+        logger.info("Enable mgmt vrf and configure IP on the mgmt interface")
+        self.configure_mgmt_vrf_test_data(mgmt_interface)
+
+        logger.info("Create data vrf and configure IP on the data vrf interface")
+        self.configure_data_vrf_test_data(routed_interfaces)
+
+    @pytest.fixture(scope="function", autouse=True)
+    def clear_syslog_config(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_frontend_asic_index):
+        """
+        clear syslog config for very test
+        """
+        self.duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+
+        yield
+        syslog_config_list = show_syslog(self.duthost)
+        if syslog_config_list:
+            for syslog_config in syslog_config_list:
+                del_syslog_server(self.duthost, syslog_server_ip=syslog_config["server ip"])
+
+    def configure_data_vrf_test_data(self, routed_interfaces):
+        """
+        Configure test data for data vrf
+        """
+        create_vrf(self.duthost, VRF_LIST[1])
+        bind_interface_to_vrf(self.asichost, VRF_LIST[1], routed_interfaces[1])
+        for k, v in DATA_VRF_IP_ADDRESSES.items():
+            self.asichost.config_ip_intf(routed_interfaces[1], DATA_VRF_IP_ADDRESSES[k]["source_ip"], "add")
+            replace_ip_neigh(self.duthost, neighbour=DATA_VRF_IP_ADDRESSES[k]["syslog_server_ip"],
+                             neigh_mac_addr=DATA_VRF_IP_ADDRESSES[k]["syslog_server_mac"],
+                             dev=VRF_LIST[1])
+
+    def configure_default_vrf_test_data(self, routed_interfaces):
+        """
+        Configure test data for default vrf
+        """
+        for k, v in DEFAULT_VRF_IP_ADDRESSES.items():
+            self.asichost.config_ip_intf(routed_interfaces[0], DEFAULT_VRF_IP_ADDRESSES[k]["source_ip"], "add")
+            replace_ip_neigh(self.duthost, neighbour=DEFAULT_VRF_IP_ADDRESSES[k]["syslog_server_ip"],
+                             neigh_mac_addr=DEFAULT_VRF_IP_ADDRESSES[k]["syslog_server_mac"],
+                             dev=routed_interfaces[0])
+
+    def configure_mgmt_vrf_test_data(self, mgmt_interface):
+        """
+        Configure test data for mgmt vrf
+        """
+        create_vrf(self.duthost, VRF_LIST[2])
+        # when create mgmt vrf, dut connection will be lost for a while
+        time.sleep(10)
```

Comment:
>10 [](http://example.com/codeflow?start=19&length=2)

How stable is 10s for recovery? @prsunny Is it a better way to wait?

Reply:
Agree, this would be unpredictable. One approach I can think of is to ping and timeout after say 3minutes.